src/thunderbots/software/ai/navigator/obstacle/obstacle.h

```diff
@@ -3,27 +3,82 @@
  */
 #pragma once
 
+#include "ai/world/ball.h"
 #include "ai/world/robot.h"
+#include "geom/angle.h"
 #include "geom/point.h"
 #include "geom/polygon.h"
 #include "geom/util.h"
+#include "shared/constants.h"
 
 class Obstacle
 {
    public:
+    static Obstacle createPlaceholderObstacle();
+
+    static Obstacle createRobotObstacle(const Robot& robot, bool enable_velocity_cushion);
+
     /*
      * Gets the boundary polygon around the given robot obstacle that other robots
-     * should not enter with a buffer scaled by radiusScaling and
-     * velocity_projection_scaling
+     * should not enter with a buffer scaled by radius_scaling and
+     * velocity_cushion_scaling
      *
      * @param robot robot to create obstacle boundary polygon around
-     * @param radiusScaling safety factor to vary the radius size of the buffer zone
-     * (1=default), must be greater than 0
-     * @param velocity_projection_scaling scales the projection buffer in the direction of
-     * the velocity (1=default), must be greater than 0
+     * @param radius_scaling multiplicatively scales the radius of the obstacle
+     * @param velocity_cushion_scaling multiplicatively scales the cushion of
+     * velocity of the robot
```

Comment:
This is a bit vague, I would certainly have to go look at the implementation to start understanding what's going on here. Could we make this a bit more specific?

Reply:
Questions that come to mind:
* What's the default value of the "cushion" value getting scaled? Is it directly proportional to the velocity, or something slightly extra?

Reply:
it will directly proportional to the velocity. I'll add more commenting to clarify

Reply:
Looks like this is unchanged?

Reply:
I added comments above this line